### PR TITLE
Enforce source checks

### PR DESCRIPTION
**Pull request by lbonn**
It's currently very easy to forget to add new source and headers to the static checks. It causes unclean files to be committed and unnecessary review back and forth.

Example: https://github.com/advancedtelematic/aktualizr/pull/620#issuecomment-371456780

CMake now tries to match the list of source files it can find with what was added to the checks.

### CODE REVIEW

File: src/opcuabridge/utility.h [hunk 1]
```diff
@@ -4,23 +4,24 @@
...
+#define SERIALIZE_FIELD_DECL_BEGIN(STREAM) \
+  template <typename T>                    \
+  struct serialize_field<STREAM, T> {      \
+  void operator()(STREAM &ar, const std::string &xml_tag, T &d)
 
-#define SERIALIZE_FIELD_DECL_END                                               \
-  };
+#define SERIALIZE_FIELD_DECL_END \
+  }                              \
+  ;
```
Comment (patriotyk) [inline src/opcuabridge/utility.h]: Remove new line.

**Comment by lbonn** [comment]
Rebased after #629

**Comment by cajun-rat** [comment]
(rebased)

File: src/sota_tools/CMakeLists.txt [hunk 2]
```diff
@@ -13,92 +13,94 @@ set (SOTA_TOOLS_LIB_SRC
   deploy.cc
 )
 
-add_library(sota_tools_static_lib STATIC ${SOTA_TOOLS_LIB_SRC})
+if (BUILD_SOTA_TOOLS)
+  add_library(sota_tools_static_lib STATIC ${SOTA_TOOLS_LIB_SRC})
+  target_compile_options(sota_tools_static_lib PUBLIC -std=c++11)
```
Comment (patrickvacek) [inline src/sota_tools/CMakeLists.txt]: This line is no longer necessary since we've moved everything to C++11, right?

**Pull request merged**